meson.build

```diff
@@ -75,6 +75,15 @@ configinc = include_directories('.')
 cc = meson.get_compiler('c')
 dl_dep = cc.find_library('dl', required : false)
 
+if cc.get_argument_syntax() == 'gcc'
+  add_project_arguments(
+    cc.get_supported_arguments([
+      '-Werror=missing-prototypes',
```

Comment:
I do not consider forcing errors to be a good practice. I suggest to add a change which will actually allow us to see warnings and add error forcing on ci level (basically we already are doing that, but we miss some warnings exposure, likely need `-Wextra`).

Reply:
> I do not consider forcing errors to be a good practice.

A number of well established projects each with very wide developer-base will disagree with you there - for example [mesa](https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/meson.build#L1178) and [systemd](https://github.com/systemd/systemd/blob/main/meson.build#L448).

The point is that you want a small set of issues to be flagged and fixed, early, on as many compiler/OS versions as possible. The CI covers a very limited set, plus it happens quite late - during PR submission.